cloud/services/virtualmachines/virtualmachines.go

```diff
@@ -243,37 +243,29 @@ func generateImageReference(image infrav1.Image) (*compute.ImageReference, error
 
 	if image.ID != nil {
 		imageRef.ID = to.StringPtr(*image.ID)
-
-		// return early since we should only need the image ID
 		return imageRef, nil
-	} else if image.SubscriptionID != nil && image.ResourceGroup != nil && image.Gallery != nil && image.Name != nil && image.Version != nil {
+	}
+
+	if image.SubscriptionID != nil && image.ResourceGroup != nil && image.Gallery != nil && image.Name != nil && image.Version != nil {
```

Comment:
Probably simplify this further.

```golang
                imageID, err := generateImageID(image)
		if err == nil {
                        imageRef.ID = to.StringPtr(imageID)

		  // return early since we're referencing an image that may not be published
		   return imageRef, nil
		}

		return generateImagePlan(image)
                 

func generateImagePlan(image <>) {
    if image.Publisher == nil  {
        return nil, errors.New("Image reference cannot be generated, as Publisher is missing")
} 
...
}
```